src/Workspaces/Core/Portable/Shared/Extensions/SourceTextExtensions.cs

```diff
@@ -17,6 +17,10 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions;
 
 internal static partial class SourceTextExtensions
 {
+    // char pooled memory : 8K * 256 = 2MB
+    private const int CharBufferSize = 4 * 1024;
+    private static readonly ObjectPool<char[]> CharArrayPool = new(() => new char[CharBufferSize], 256);
```

Comment:
moved these here as this is the only place it is used.  and it is easier to validate proper usage.